cpp/src/interpolate/cubic_spline.cu

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial/cubic_spline.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/filling.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include "cusparse.h"
+#include <cuspatial/utility.hpp>
+
+namespace { // anonymous
+
+struct parallel_search {
```

Comment:
Docs.

Reply:
What docs do you require for a .cu file? Just a /** block saying what this does?

Reply:
That would be nice. Doesn't have to be `/**` (doxygen), can be `//` -- just for readers of the code.